src/test/kotlin/game/testFactory/WarriorFactory.kt

```diff
@@ -0,0 +1,24 @@
+package game.testFactory
+
+import game.characters.*
+import game.interfaces.BaseWarrior
+import game.testClass.Rookie
+import java.util.*
+
+
+fun  getWarrior(type : String) :  BaseWarrior {
+         var warrior : BaseWarrior? = null
+    if (type.isNotEmpty() and type.isNotBlank()) {
+        when (type.lowercase(Locale.getDefault())) {
+            "knight" -> warrior = Knight(Warrior())
+            "vampire" -> warrior = Vampire(Warrior())
+            "warrior" -> warrior = Warrior()
+            "lancer" -> warrior = Lancer(Warrior())
+            "defender" -> warrior = Defender(Warrior())
+            "rookie" -> warrior = Rookie(Warrior())
+        }
+    }
+
+    return  warrior!!
```

Comment:
please remove force unwrap + we need enum

Reply:
Changed to the new implementation